### DISCUSSION TITLE
Sorting table with data from API request seems to be broken

### DISCUSSION THREAD

**Discussion by crsnchng** [Budibase/budibase#5757] [Help]
Edit: @mjashanks, Reopening - It may just be me but during my testing today, I've noticed a number of inconsistencies with sorting. In a table where I have some names and numbers, all of the names sort correctly whether viewing on the table component or in the back end Budibase DB viewer. 

Numbers, however, seem to be hit or miss. Some larger, 6 digit numbers, _seem_ to sort just fine but single digit to 3 digit spreads of numbers seem to be following a hidden separator. From a table component with a Budibase DB data source, when I click the column title, it seems like groups of them are being sorted. A handful of them are sorted correctly and then a high number outlier comes along but everything below that outlier until the next outlier is sorted correctly. **The only time this column sorts correctly on the table component is when I select the column as the default sort.** 

In addition to this, the same Budibase DB table sorts names and text just fine, as previously mentioned, but REST API request data seems to use different sorting. For instance, if we have two text values: "Maverick" and "mAverick", the API will sort the former with the other capital M's but the latter falls to the bottom/top of the sort as if the lower case is being treated like a special character.

The below/initial issue also still persists. The schemas are correct and the API that I am querying is most definitely returning a number (I added the following to the object I'm returning in my transformer:  `test: typeof(valueInQuestion)` and it returned "number")

Best,

C

------------------------------------------------------------------------------------------------------------------------------------------------

Hello,

When I have a table displaying data with correctly defined schema (number, text, etc. from the schema definitions in the query settings), the number columns seem to be sorting wrong. For example, it will be: 

- 99
- 95
- 9
- 83
- 8
- 73
- 7.6
- 64

Could I potentially be doing something incorrectly here?

Thanks in advance.

**Comment by mjashanks** [comment]
Hey @crsnchng 

The table is sorting it alphabetically, which suggests that it's being converted to a string at some point. Can you check that 
1. Your API is returning a number like `my_num: 6` and not a string, like `my_num: "6"`
2. Go to your REST query in budibase, and check the schema. Your field should be "Number", not "Text"

**Reply by crsnchng** [reply]
> Hey @crsnchng
> 
> The table is sorting it alphabetically, which suggests that it's being converted to a string at some point. Can you check that
> 
>     1. Your API is returning a number like `my_num: 6` and not a string, like `my_num: "6"`
> 
>     2. Go to your REST query in budibase, and check the schema. Your field should be "Number", not "Text"

Hi Michael,

Thanks for the reply. The API is returning an integer and the schema is set to number. Seems like if it is being converted, it is occurring between the API query and it being displayed on the table.

Best,

C

**Reply by crsnchng** [reply]
Just noticed that the "Preview" area of the Query is able to sort the values correctly but the table on the front end seems to be failing.

Edit: When trying to test using the built in Table block vs data source + table, all API queries return 500 Internal Server Error. Seems like there may be a deeper issue here.

Best,

C

**Reply by crsnchng** [reply]
@mjashanks,

I have been messing with this all day and I think all my issues are resolved at this point. I think it is a mix of pagination and potentially bad data.

I will mark this as resolved while I keep testing.

Best,

C

**Reply by crsnchng** [reply]
@mjashanks 

This issue persists in my installation with various values. Per my edit above, all the issues mentioned in the edit still persist. I have confirmed schemas are correct and values are set correctly. I have no idea what could be causing this.

Best,

C
